tests/D2L.CodeStyle.Analyzers.Test/Async/Generator/AsyncToSyncMethodTransformerTests.cs

```diff
@@ -319,6 +328,43 @@ internal sealed class AsyncToSyncMethodTransformerTests {
 		Assert.AreEqual( @"[Blocking] Foo Bar() { var ct = new CancellationToken(); return Foo( 1,""test"" ); }", actual.Value.ToFullString() );
 	}
 
+	[Test]
+	public void ParenthesizedAnonymousCreationLambda() {
+		var actual = Transform( @"[GenerateSync] async Task BarAsync() { await BazAsync(() => new { BazAsync = 5, Quux = ""test"" } ); }" );
+
+		Assert.IsTrue( actual.Success );
+		Assert.IsEmpty( actual.Diagnostics );
+		// For now we are not transforming the variable names
+		Assert.AreEqual( @"[Blocking] void Bar() { Baz(() => new { BazAsync = 5,Quux = ""test"" } ); }", actual.Value.ToFullString() );
+	}
+
+	[Test]
+	public void AnonymousCreation() {
+		var actual = Transform( @"[GenerateSync] async Task BarAsync() { var options = new { size = new { width = GetCurrentWidthAsync(), height = 200 } }; await JsonSerializer.SerializeAsync(options); }" );

```

Comment:
There is no `await` on `GetCurrentWidthAsync()` here, which would presumably have some `Task<T>` type from the context.

The rewriter is turning it into a call to `GetCurrentWidth()` which you'd imagine would have type `T`, at least not `Task<T>`.... so the type of `size` would be different in the rewritten code.

I'm just observing here, not implying something should be different. It's OK but I think this is a case where the generator is doing something that will generate compiler errors... which is OK, not the worst thing.

If we put an `await` in the test then the test might be demonstrating the utility of the generator better, even if it would obscure the quirk.

Reply:
Good observation. I think I'll leave it as is for now